### ISSUE TITLE
Lint idea: Collection is only added to, never read

### ISSUE THREAD

**Issue by schubart** [rust-lang/rust-clippy#9267]
### What it does

Adding to a collection and never using it is pointless and might indicate that the developer forgot to do something with the collection.

Currently this does not raise warnings:

```rust
#![warn(clippy::all, clippy::pedantic)]

fn main() {
    let mut numbers = Vec::new();
    for number in 1..10 {
        numbers.push(number);
    }
}
```

The new lint would point out that `numbers` is never read. 

To fix the warning, one could delete all code in `main` or do something with `numbers`, such as printing it:

```
#![warn(clippy::all, clippy::pedantic)]

fn main() {
    let mut numbers = Vec::new();
    for number in 1..10 {
        numbers.push(number);
    }
}
```

What do people think? If this sounds useful, I'd like to do it as my first contribution. 

### Lint Name

collection_is_never_read

### Category

suspicious

### Advantage

The new code no longer unnecessarily does the work of updating the collection.

### Example

```rust
<code>
```

Could be written as:

```rust
#![warn(clippy::all, clippy::pedantic)]

fn main() {
    let mut numbers = Vec::new();
    for number in 1..10 {
        numbers.push(number);
    }
}
```

**Comment by llogiq** [comment]
Note that this lint should only apply to sets if the `insert` result is unused.
